src/ILCompiler.Compiler/src/Compiler/DependencyAnalysis/NodeFactory.cs

```diff
@@ -283,6 +297,21 @@ public ISymbolNode HelperEntrypoint(HelperEntrypoint entrypoint)
             return symbol;
         }
 
+        private TypeDesc _systemICastableType;
+
+        public TypeDesc ICastableInterface
```

Comment:
As of right now, the factory is dumping ground for runtime-specific datastructure building. Cached _typeOptionalFields and cached _systemICastableType are both very specific to the current EEType content - belong together in my mind.